lib/utils.js

```diff
@@ -123,23 +123,27 @@ function authorizedRequestFactory(scopes) {
 }
 
 /**
- * Attempts to retrieve the project number for the current active project.
- * If the project number cannot be discovered locally from the
- * GCLOUD_PROJECT_NUM environment variable, it will attempt to retrieve
- * the number from remote metadata.
+ * Attempts to retrieve the project number for the current active project from
+ * the metadata service (See https://cloud.google.com/compute/docs/metadata).
  *
+ * @param {object=} headers optional headers to include in the http request.
+ *     Note that the headers, if provided, may be extended with extra
+ *     properties.
  * @param {function(?, number):?} callback an (err, result) style callback
  */
-function getProjectNumber(callback) {
-  if (process.env.GCLOUD_PROJECT_NUM) {
-    callback(null, process.env.GCLOUD_PROJECT_NUM);
-    return;
+function getProjectNumber(headers, callback) {
+  if (typeof headers === 'function') {
+    callback = headers;
+    headers = {};
   }
 
+  headers['Metadata-Flavor'] = 'Google';
+
   var request = require('request');
   requestWithRetry(request, {
-    url: 'http://metadata/computeMetadata/v1/project/numeric-project-id',
-    headers: { 'Metadata-Flavor': 'Google'},
+    url: 'http://metadata.google.internal/computeMetadata/v1/project/' +
```

Comment:
Good idea in general, but this one I am hesitant to move because I want to extract utils into a separate module.